From 9b34149d25c3ed9dc91204c4944e102a487e5245 Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Thu, 13 Jun 2024 15:00:45 +0200
Subject: [PATCH] dt-bindings: display: Add Youritech ILI9806E panel bindings

Add the bindings for the MIPI-DSI display from Youritech with the
ILI9806E controller.

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
---
 .../display/panel/youritech,ili9806e.yaml     | 84 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 85 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml

diff --git a/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml b/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
new file mode 100644
index 000000000000..9c5eb2bf2e1f
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/youritech,ili9806e.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: MIPI-DSI panel with Youritech ILI9806E Controller
+
+maintainers:
+  - Daniel Ammann <daniel.ammann@bytesatwork.ch>
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    const: youritech,ili9806e
+
+  reg: true
+  port: true
+  reset-gpios: true
+  width-mm: true
+  height-mm: true
+
+  dsi-lanes:
+    description: Number of DSI lanes to be used must be <1> or <2> or <3> or <4>
+    enum: [1, 2, 3, 4]
+
+  video-mode:
+    description: |
+      0 - burst-mode
+      1 - non-burst with sync event
+      2 - non-burst with sync pulse
+      3 - command mode
+    $ref: /schemas/types.yaml#/definitions/uint32
+    enum: [0, 1, 2, 3]
+
+required:
+  - compatible
+  - reg
+  - dsi-lanes
+  - port
+
+additionalProperties: false
+
+examples:
+  - |
+    &dsi {
+      #address-cells = <1>;
+      #size-cells = <0>;
+      status = "okay";
+
+      panel@0 {
+        compatible = "youritech,ili9806e";
+        reg = <0>;
+        backlight = <&mipi_bl>;
+        reset-gpios = <&gpio2 22 GPIO_ACTIVE_HIGH>;
+        dsi-lanes = <2>;
+        video-mode = <2>;	/* 0: burst mode
+              * 1: non-burst mode with sync event
+              * 2: non-burst mode with sync pulse
+              */
+        width-mm = <68>;
+        height-mm = <121>;
+        status = "okay";
+
+        port {
+          panel_mipi_in: endpoint {
+            remote-endpoint = <&dsi_out>;
+          };
+        };
+      };
+
+      ports {
+        port@1 {
+          reg = <1>;
+
+          dsi_out: endpoint {
+            remote-endpoint = <&panel_mipi_in>;
+          };
+        };
+      };
+    };
+...
diff --git a/MAINTAINERS b/MAINTAINERS
index fe8ac3bb4156..3cd1006128b1 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -4442,6 +4442,7 @@ M:	Daniel Ammann <daniel.ammann@bytesatwork.ch>
 S:	Maintained
 F:	.github/workflows/checkpatch.yml
 F:	Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
+F:	Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
 F:	drivers/gpu/drm/panel/panel-youritech-ili9806e.c
 
 C-MEDIA CMI8788 DRIVER
